gma/src/ios/native_ad_image_ios.mm

```diff
@@ -103,5 +155,27 @@
   return internal_->scale;
 }
 
+/// Gets the auto loaded image as a vector of bytes.
+const std::vector<unsigned char> NativeAdImage::image() const {
```

Comment:
This method blocks, correct? If so it should return a future instead of a vector. Alternatively you could download the image during the ad load process instead, and simply just return the data here.

Reply:
Thanks David! Updated it to return a future and process the image loading asynchronously, rather than as a blocking call.